pkg/router/controller/unique_host.go

```diff
@@ -131,6 +131,24 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 				}
 			}
 			if !added {
+				// Clean out any old form of this route
+				next := []*routeapi.Route{}
+				for i := range old {
+					if old[i].Name != route.Name {
```

Comment:
Yeah, at line 122 it screens the list by the namespace of the oldest path.  I saw that check and thought about it, but I think it is redundant.